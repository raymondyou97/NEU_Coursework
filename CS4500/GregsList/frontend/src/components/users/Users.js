import React, { Component } from 'react';

import '../../styles/users.css';

class Users extends Component {
  componentDidMount() {
    this.props.fetchUsers();
  }

  updateUsername = e => {
    this.props.updateUsername(e.target.value);
  };

  updateFirstName = e => {
    this.props.updateFirstName(e.target.value);
  };

  updateLastName = e => {
    this.props.updateLastName(e.target.value);
  };

  updatePassword = e => {
    this.props.updatePassword(e.target.value);
  };

  deleteUser = id => {
    this.props.deleteUser(id);
  };

  editUser = (id, username, firstName, lastName, password) => {
    this.props.selectUser(id);
    this.props.updateUsername(username);
    this.props.updateFirstName(firstName);
    this.props.updateLastName(lastName);
    this.props.updatePassword(password);
  };

  render() {
    const { selectedUser } = this.props;

    return (
      <div>
        <h3 className="user-title">Users</h3>
        <table className="user-table">
          <tbody className="user-table-body">
            {/* This is the name of the columns */}
            <tr className="user-columns">
              <td>Username</td>
              <td>First Name</td>
              <td>Last Name</td>
              <td>Password</td>
              <td></td>
            </tr>
            {/* This is the input fields for the columns */}
            <tr className="user-inputs">
              <td>
                <input
                  className="user-input-username"
                  type="text"
                  onChange={this.updateUsername}
                  placeholder="Username"
                  value={selectedUser.username}
                />
              </td>
              <td>
                <input
                  className="user-input-firstname"
                  type="text"
                  onChange={this.updateFirstName}
                  placeholder="First Name"
                  value={selectedUser.firstName}
                />
              </td>
              <td>
                <input
                  className="user-input-lastname"
                  type="text"
                  onChange={this.updateLastName}
                  placeholder="Last Name"
                  value={selectedUser.lastName}
                />
              </td>
              <td>
                <input
                  className="user-input-password"
                  type="text"
                  onChange={this.updatePassword}
                  placeholder="Password"
                  value={selectedUser.password}
                />
              </td>
              <td className="user-input-buttons">
                <button
                  className="user-add-button"
                  onClick={this.props.createUser}
                >
                  Add
                </button>
                <button
                  className="user-update-button"
                  onClick={this.props.updateUser}
                >
                  Update
                </button>
              </td>
            </tr>
            {/* This loops through the users */}
            {this.props.users.map(user => {
              const { id, username, firstName, lastName, password } = user;
              return (
                <tr className="user-details" key={id}>
                  <td onClick={() => this.props.showDetails(id)}>{username}</td>
                  <td onClick={() => this.props.showDetails(id)}>
                    {firstName}
                  </td>
                  <td onClick={() => this.props.showDetails(id)}>{lastName}</td>
                  <td onClick={() => this.props.showDetails(id)}>{password}</td>
                  <td className="user-buttons">
                    <button
                      className="user-delete-button"
                      onClick={() => this.deleteUser(id)}
                    >
                      Delete
                    </button>
                    <button
                      className="user-edit-button"
                      onClick={() =>
                        this.editUser(
                          id,
                          username,
                          firstName,
                          lastName,
                          password
                        )
                      }
                    >
                      Edit
                    </button>
                  </td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </div>
    );
  }
}

export default Users;
