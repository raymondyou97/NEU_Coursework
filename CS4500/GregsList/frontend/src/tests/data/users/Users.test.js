'use es6';

import React from 'react';
import Users from '../../../components/users/Users';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { USER_LIST } from './UserData';

test('Snapshot renders', () => {
  const selectedUser = {
    username: '',
    firstName: '',
    lastName: '',
  };

  const component = renderer.create(
    <Users
      fetchUsers={() => {}}
      selectedUser={selectedUser}
      selectUser={() => {}}
      users={USER_LIST}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      createUser={() => {}}
      showDetails={() => {}}
      deleteUser={() => {}}
      updateUser={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Users component updates when there is user data', () => {
  const component = renderer.create(
    <Users
      fetchUsers={() => {}}
      selectedUser={{
        username: 'spoonily',
        firstName: 'Jonah',
        lastName: 'Min',
      }}
      selectUser={() => {}}
      users={USER_LIST}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      createUser={() => {}}
      showDetails={() => {}}
      deleteUser={() => {}}
      updateUser={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Initial DOM contains correct number of rows', () => {
  const selectedUser = {
    username: '',
    firstName: '',
    lastName: '',
  };

  const component = shallow(
    <Users
      fetchUsers={() => {}}
      selectedUser={selectedUser}
      selectUser={() => {}}
      users={USER_LIST}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      createUser={() => {}}
      showDetails={() => {}}
      deleteUser={() => {}}
      updateUser={() => {}}
    />
  );

  expect(component.find('tr').length).toEqual(5);
});

test('Users component input value changes if selected user specified', () => {
  const selectedUser = {
    username: 'spoonily',
    firstName: 'Jonah',
    lastName: 'Min',
  };

  const component = shallow(
    <Users
      fetchUsers={() => {}}
      selectedUser={selectedUser}
      selectUser={() => {}}
      users={USER_LIST}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      createUser={() => {}}
      showDetails={() => {}}
      deleteUser={() => {}}
      updateUser={() => {}}
    />
  );

  expect(component.find('.user-input-username').props().value).toEqual(
    'spoonily'
  );
  expect(component.find('.user-input-firstname').props().value).toEqual(
    'Jonah'
  );
  expect(component.find('.user-input-lastname').props().value).toEqual('Min');
});
