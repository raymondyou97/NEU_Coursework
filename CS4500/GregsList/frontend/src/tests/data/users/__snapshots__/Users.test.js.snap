// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders 1`] = `
<div>
  <h3
    className="user-title"
  >
    Users
  </h3>
  <table
    className="user-table"
  >
    <tbody
      className="user-table-body"
    >
      <tr
        className="user-columns"
      >
        <td>
          Username
        </td>
        <td>
          First Name
        </td>
        <td>
          Last Name
        </td>
        <td>
          Password
        </td>
        <td />
      </tr>
      <tr
        className="user-inputs"
      >
        <td>
          <input
            className="user-input-username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
        </td>
        <td>
          <input
            className="user-input-firstname"
            onChange={[Function]}
            placeholder="First Name"
            type="text"
            value=""
          />
        </td>
        <td>
          <input
            className="user-input-lastname"
            onChange={[Function]}
            placeholder="Last Name"
            type="text"
            value=""
          />
        </td>
        <td>
          <input
            className="user-input-password"
            onChange={[Function]}
            placeholder="Password"
            type="text"
          />
        </td>
        <td
          className="user-input-buttons"
        >
          <button
            className="user-add-button"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="user-update-button"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="user-details"
      >
        <td
          onClick={[Function]}
        >
          orchy
        </td>
        <td
          onClick={[Function]}
        >
          David
        </td>
        <td
          onClick={[Function]}
        >
          Smithu
        </td>
        <td
          onClick={[Function]}
        />
        <td
          className="user-buttons"
        >
          <button
            className="user-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="user-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="user-details"
      >
        <td
          onClick={[Function]}
        >
          kelvin
        </td>
        <td
          onClick={[Function]}
        >
          kelvin
        </td>
        <td
          onClick={[Function]}
        >
          deleon
        </td>
        <td
          onClick={[Function]}
        />
        <td
          className="user-buttons"
        >
          <button
            className="user-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="user-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="user-details"
      >
        <td
          onClick={[Function]}
        >
          jonah
        </td>
        <td
          onClick={[Function]}
        >
          Jonah
        </td>
        <td
          onClick={[Function]}
        >
          MIn
        </td>
        <td
          onClick={[Function]}
        />
        <td
          className="user-buttons"
        >
          <button
            className="user-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="user-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Users component updates when there is user data 1`] = `
<div>
  <h3
    className="user-title"
  >
    Users
  </h3>
  <table
    className="user-table"
  >
    <tbody
      className="user-table-body"
    >
      <tr
        className="user-columns"
      >
        <td>
          Username
        </td>
        <td>
          First Name
        </td>
        <td>
          Last Name
        </td>
        <td>
          Password
        </td>
        <td />
      </tr>
      <tr
        className="user-inputs"
      >
        <td>
          <input
            className="user-input-username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value="spoonily"
          />
        </td>
        <td>
          <input
            className="user-input-firstname"
            onChange={[Function]}
            placeholder="First Name"
            type="text"
            value="Jonah"
          />
        </td>
        <td>
          <input
            className="user-input-lastname"
            onChange={[Function]}
            placeholder="Last Name"
            type="text"
            value="Min"
          />
        </td>
        <td>
          <input
            className="user-input-password"
            onChange={[Function]}
            placeholder="Password"
            type="text"
          />
        </td>
        <td
          className="user-input-buttons"
        >
          <button
            className="user-add-button"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="user-update-button"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="user-details"
      >
        <td
          onClick={[Function]}
        >
          orchy
        </td>
        <td
          onClick={[Function]}
        >
          David
        </td>
        <td
          onClick={[Function]}
        >
          Smithu
        </td>
        <td
          onClick={[Function]}
        />
        <td
          className="user-buttons"
        >
          <button
            className="user-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="user-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="user-details"
      >
        <td
          onClick={[Function]}
        >
          kelvin
        </td>
        <td
          onClick={[Function]}
        >
          kelvin
        </td>
        <td
          onClick={[Function]}
        >
          deleon
        </td>
        <td
          onClick={[Function]}
        />
        <td
          className="user-buttons"
        >
          <button
            className="user-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="user-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="user-details"
      >
        <td
          onClick={[Function]}
        >
          jonah
        </td>
        <td
          onClick={[Function]}
        >
          Jonah
        </td>
        <td
          onClick={[Function]}
        >
          MIn
        </td>
        <td
          onClick={[Function]}
        />
        <td
          className="user-buttons"
        >
          <button
            className="user-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="user-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
